src/client/server_result.ts

```diff
@@ -0,0 +1,86 @@
+        this.request_log = request_log;
+        this.atf_content = ""; //placeholder for lemmatisation
+    }
+
+    get_validation_errors(oracc_log: string){
+        /* Process the oracc log and extract error line numbers and error
+        messages to build a dictionary the front end can use.
+        The purpose of this is to show different styling in those line numbers
+        and potentially a tooltip that shows the error message.
+        */
+        // TODO for some reason the keys are being converted to string, why?
+        const validation_errors: { [line_num: number]: string } = {};
+        let lines = oracc_log.split(os.EOL);
+        // Split os.EOL will always add an empty string at the end, so we have
+        // to remove it manually
+        lines.pop();
+        for (const line of lines){
+            // Check if this is an error line or the error summary
+            if (line.includes(':')){
+                // TODO wrap in try/catch in case we get an error when
+                // trying to access [1] and [0]
+                const line_number = Number(line.split(':')[1]);
+                // Sometimes error messages contain ":", so we can't take the
+                // last element in the split array. We need to join all
+                // elements after the 3rd occurrence of ":".
+                const position = line.split(":", 3).join(":").length;
+                const error_message = line.substr(position + 2);
+                //Note more than one error can exist per line
+                if (line_number in validation_errors){
+                    // TODO We can do this in a more elegant way with the
+                    // values being string[] rather than everything in one
+                    // string.
+                    validation_errors[line_number] += ". " + error_message;
+                }
+                else {
+                    validation_errors[line_number] = error_message;
+                }
+            }
+            else {
+                // Save summary line as line 0
+                validation_errors[0] = line;
+            }
+        }
+        return validation_errors;
+    }
+
+    get_user_log(){
+        /* This method will build the log they user will see in the VS code
+           console */
+        if (Object.keys(this.validation_errors).length == 0){
+            return "ATF validation returned no errors.";
+        }
+        else {
+            let user_log = "";
+            for (const line_num in this.validation_errors) {
+                // dict key 0 contains the summary, so skip
+                if (Number(line_num) != 0){
+                    const error_msg = this.validation_errors[line_num];
+                    //TODO check if \n is valid here or we need <br>
+                    user_log += `Line ${line_num}: ${error_msg}.${os.EOL}`;
+                }
+            }
+            // Add summary line at the end
+            user_log += this.validation_errors[0];
```

Comment:
I only observed them being string in the tests. Maybe they get converted to string when printed out? That still makes no sense. I'm not sure. This line was working fine, but now it's been replace with `this.summary_line`.

Reply:
More confusion:

```
src/client/server_result.ts:78:52 - error TS2367: This condition will always return 'true' since the types 'string' and 'number' have no overlap.

78                 .filter(([line_num, error_msg]) => line_num != 0)
                                                      ~~~~~~~~~~~~~


Found 1 error.

```